src/ImageSharp/Formats/Tiff/Compression/Compressors/DeflateCompressor.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.IO;
+using SixLabors.ImageSharp.Compression.Zlib;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Constants;
+using SixLabors.ImageSharp.Memory;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff.Compression.Compressors
+{
+    internal class DeflateCompressor : TiffBaseCompressor
+    {
+        private readonly DeflateCompressionLevel compressionLevel;
+
+        private readonly MemoryStream memoryStream = new MemoryStream();
+
+        public DeflateCompressor(Stream output, MemoryAllocator allocator, int width, int bitsPerPixel, TiffPredictor predictor, DeflateCompressionLevel compressionLevel)
+            : base(output, allocator, width, bitsPerPixel, predictor)
+            => this.compressionLevel = compressionLevel;
+
+        /// <inheritdoc/>
+        public override TiffEncoderCompression Method => TiffEncoderCompression.Deflate;
+
+        /// <inheritdoc/>
+        public override void Initialize(int rowsPerStrip)
+        {
+        }
+
+        /// <inheritdoc/>
+        public override void CompressStrip(Span<byte> rows, int height)
+        {
+            this.memoryStream.Seek(0, SeekOrigin.Begin);
+            using var stream = new ZlibDeflateStream(this.Allocator, this.memoryStream, this.compressionLevel);
+
+            if (this.Predictor == TiffPredictor.Horizontal)
+            {
+                HorizontalPredictor.ApplyHorizontalPrediction(rows, this.BytesPerRow, this.BitsPerPixel);
+            }
+
+            stream.Write(rows);
+
+            stream.Flush();
+            stream.Dispose();
```

Comment:
You're calling Dispose 2x here.

Reply:
yes, it seems `Flush` is excess, tests are pass without it

Reply:
i have removed the duplicate Dispose (b0e965fd), keeping the Flush as i dont think its not needed.

Reply:
@brianpopow `Dispose` method write `CRC`, I think it would be more correct to use `Dispose`

Reply:
@IldarKhayrutdinov It is calling Dispose. It looks like this now:

```
using (var stream = new ZlibDeflateStream(this.Allocator, this.memoryStream, this.compressionLevel))
{
    if (this.Predictor == TiffPredictor.Horizontal)
    {
        HorizontalPredictor.ApplyHorizontalPrediction(rows, this.BytesPerRow, this.BitsPerPixel);
    }

    stream.Write(rows);
    stream.Flush();
}
```

When the using scope is left, dispose will be called.

Reply:
@brianpopow oh yeah pardon, I didn't see it